src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/abi/CallingSequenceBuilder.java

```diff
@@ -90,7 +90,12 @@ private static void verifyUnboxBindings(Class<?> inType, List<Binding> bindings)
         stack.push(inType);
 
         for (Binding b : bindings) {
-            b.verifyUnbox(stack);
+            if (b.tag() == Binding.Tag.VM_LOAD
```

Comment:
We could have static EnumSet for what's an box vs. unbox binding, so that you can filter by checking for presence in the right enum set?

Reply:
Yeah, that's a good idea.